pkg/cluster/machine/machine.go

```diff
@@ -43,13 +39,15 @@ func IsNode(machine *clusterv1.Machine) bool {
 
 // FirstMaster scans the provided array of machines and return the first
 // one which is a "Master" or nil if none.
-func FirstMaster(machines []*clusterv1.Machine) *clusterv1.Machine {
-	for _, machine := range machines {
+// Machines and BareMetalMachines must be in the same order
+func FirstMaster(machines []*clusterv1.Machine, bl []*baremetalspecv1.BareMetalMachine) (*clusterv1.Machine, *baremetalspecv1.BareMetalMachine) {
```

Comment:
I wonder if we should rethink how we denote 'master` nodes. I'm not sure a label on `Machine` is the right place. I think we should probably implement a capi control plane instead to make it explicit. https://cluster-api.sigs.k8s.io/developer/architecture/controllers/control-plane.html

Reply:
Assuming there is no way to reuse the existing kubeadm control plane. Or perhaps support both?